crates/llms/src/chat/mistral.rs

```diff
@@ -25,13 +27,16 @@ use mistralrs::{
 };
 use mistralrs_core::{LocalModelPaths, ModelPaths, Pipeline};
 use snafu::ResultExt;
-use std::{path::Path, str::FromStr, sync::Arc};
-use tokio::sync::mpsc::{channel, Receiver, Sender};
+use std::{path::Path, pin::Pin, str::FromStr, sync::Arc};
+use tokio::sync::{
+    mpsc::{channel, Receiver, Sender},
+    RwLock,
+};
 
 pub struct MistralLlama {
     pipeline: Arc<MistralRs>,
     tx: Sender<MistralRsponse>,
-    rx: Receiver<MistralRsponse>,
+    rx: Arc<RwLock<Receiver<MistralRsponse>>>,
```

Comment:
Instead of keeping a single channel for all requests, let's remove the channels from the struct members and create a new channel pair on each request - which will keep it isolated for multiple requests.